mlflow/tracing/config.py

```diff
@@ -0,0 +1,112 @@
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Callable, Optional
+
+from mlflow.tracing.utils.processor import validate_span_processor
+from mlflow.utils.annotations import experimental
+
+
+@dataclass
+class TracingConfig:
+    """Configuration for MLflow tracing behavior."""
+
+    # TODO: Move more configuration options here, such as async logging, display, etc.
+    # A list of functions to process spans before export.
+    span_processors: Optional[list[Callable]] = None
+
+    def __post_init__(self):
+        self.span_processors = validate_span_processor(self.span_processors)
+
+
+# Global configuration instance for tracing
+_MLFLOW_TRACING_CONFIG = TracingConfig()
+
+# Sentinel object to detect unspecified arguments
+_UNSPECIFIED = object()
+
+
+class TracingConfigContext:
+    """Context manager for temporary tracing configuration changes."""
+
+    def __init__(self, config_updates):
+        self.config_updates = config_updates
+        self.previous_config = None
+        self.is_context_manager = False
+
+        # Save the config state before applying any changes
+        self.previous_config = deepcopy(_MLFLOW_TRACING_CONFIG)
+
+        # Apply changes immediately for function-style usage
+        for key, value in self.config_updates.items():
+            setattr(_MLFLOW_TRACING_CONFIG, key, value)
+
+    def __enter__(self):
+        # Mark as context manager
+        self.is_context_manager = True
+        # Changes are already applied from __init__
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        global _MLFLOW_TRACING_CONFIG
+        # Only restore if actually used as context manager
+        if self.is_context_manager and self.previous_config is not None:
```

Comment:
We don't need this check as well? _MLFLOW_TRACING_CONFIG should always be a TracingConfig object, so previous_config can't be None?